dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -922,6 +922,34 @@ private void verifyChecksumsAreSaved(WorkflowVersion version) {
         }
     }
 
+    @Test
+    public void testChecksumsForSourceFiles() {
+        // Test grabbing checksum on refresh
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowsApi = new WorkflowsApi(webClient);
+        Workflow workflow = workflowsApi.manualRegister("github", "DockstoreTestUser2/hello-dockstore-workflow", "/Dockstore.wdl", "", "wdl", "/test.json");
+
+        workflow = workflowsApi.refresh(workflow.getId());
+        List<WorkflowVersion> workflowVersions = workflow.getWorkflowVersions();
+        assertFalse(workflowVersions.isEmpty());
+        boolean testedWDL = false;
+
+        for (WorkflowVersion workflowVersion : workflowVersions) {
+            if (workflowVersion.getName().equals("testBoth") || workflowVersion.getName().equals("testWDL")) {
+                testedWDL = true;
+                assertNotNull(workflowVersion.getSourceFiles());
+                workflowVersion.getSourceFiles().stream().forEach(sourceFile -> assertFalse("Source file should have a checksum", sourceFile.getChecksums().get(0).toString().isEmpty()));
+            }
+            workflowVersion.getSourceFiles();
+        }
+        assertTrue(testedWDL);
+
+        // Test grabbing checksum on snapshot
```

Comment:
It's probably overkill, but it would nice to verify the checksums didn't exist before doing the snapshot.